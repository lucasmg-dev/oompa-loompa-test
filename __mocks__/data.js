module.exports = {
  current: 1,
  total: 20,
  results: [
    {
      first_name: 'Marcy',
      last_name: 'Karadzas',
      favorite: {
        color: 'red',
        food: 'Chocolat'
      },
      random_string: '',
      song: '',
      gender: 'F',
      image: 'https://s3.eu-central-1.amazonaws.com/napptilus/level-test/1.jpg',
      profession: 'Developer',
      email: 'mkaradzas1@visualengin.com',
      age: 21,
      country: 'Loompalandia',
      height: 99,
      id: 1,
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum'
    },
    {
      first_name: 'Evangelia',
      last_name: 'Cowin',
      favorite: {
        color: 'red',
        food: 'cocoa nuts'
      },
      random_string: '',
      song: '',
      gender: 'M',
      image: 'https://s3.eu-central-1.amazonaws.com/napptilus/level-test/2.jpg',
      profession: 'Metalworker',
      email: 'ecowin2@visualengin.com',
      age: 22,
      country: 'Loompalandia',
      height: 96,
      id: 2,
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum'
    }
  ]
}
