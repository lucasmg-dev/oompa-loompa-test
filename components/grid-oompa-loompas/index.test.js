import '@testing-library/jest-dom'

import { GridOompaLoompas } from './index'
import { render } from '@testing-library/react'

import data from '../../__mocks__/data'

it('renders correctly', () => {
  const { asFragment } = render(<GridOompaLoompas data={data} />)
  expect(asFragment()).toMatchSnapshot()
})
