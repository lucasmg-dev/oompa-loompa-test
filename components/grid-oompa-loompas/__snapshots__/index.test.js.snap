// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <section>
    <div
      class="wrapper"
    >
      <div
        class="grid"
      >
        <article>
          <figure
            class="figure"
          >
            <img
              class="image"
              src="https://s3.eu-central-1.amazonaws.com/napptilus/level-test/1.jpg"
            />
          </figure>
          <h2
            class="name"
          >
            Marcy Karadzas
          </h2>
          <span
            class="gender"
          >
            Woman
          </span>
          <h3
            class="profession"
          >
            Developer
          </h3>
        </article>
        <article>
          <figure
            class="figure"
          >
            <img
              class="image"
              src="https://s3.eu-central-1.amazonaws.com/napptilus/level-test/2.jpg"
            />
          </figure>
          <h2
            class="name"
          >
            Evangelia Cowin
          </h2>
          <span
            class="gender"
          >
            Man
          </span>
          <h3
            class="profession"
          >
            Metalworker
          </h3>
        </article>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
